lib/lightning_web/live/run_live/index.ex

```diff
@@ -321,28 +321,24 @@ defmodule LightningWeb.RunLive.Index do
           "workorder_id" => workorder_id,
           "selected" => selected?
         },
-        %{assigns: assigns} = socket
+        socket
       ) do
+    %{page: page, selected_work_orders: selected_work_orders} = socket.assigns
+
     selected? = String.to_existing_atom(selected?)
+    workorder = Enum.find(page.entries, &(&1.id == workorder_id))
 
     work_orders =
-      if selected? do
-        workorder =
-          Enum.find(assigns.page.entries, fn wo ->
-            wo.id == workorder_id
-          end)
-
+      if selected? and nil != workorder do
```

Comment:
Wondering if we need this double condition as `nil` evaluates to false. I mean couldn't we write the condition like this: `if selected? and workorder do` ?

Reply:
It's not a double condition, it's just a more readable expression `if selected? and nil != workorder do` instead of `if selected? and nil != workorder do` calls the attention to the desired check.

Reply:
> `nil != workorder` is not a double condition, it just composes a more readable expression where `if selected? and nil != workorder do` instead of `if selected? and workorder do` calls the attention to the desired check.

Thanks for clarifying this Roger. Looks good